core/src/avm2/object/xml_object.rs

```diff
@@ -328,50 +360,84 @@ impl<'gc> TObject<'gc> for XmlObject<'gc> {
             return Ok(());
         }
 
-        let self_node = *self.node();
-        let write = self.0.write(mc);
-        let mut kind = write.node.kind_mut(mc);
-        let E4XNodeKind::Element { children, .. } = &mut *kind else {
+        // 7. Let isValidName be the result of calling the function isXMLName (section 13.1.2.1) with argument n
+        let is_valid_name = name.local_name().map_or(Ok(false), |x| {
+            crate::avm2::e4x::is_xml_name(activation, x.into())
```

Comment:
(in retrospect, maybe passing an AvmString to `is_xml_name` would have been a nicer design, so it wouldn't need a `Result` here)

Reply:
Yeah, I also realized this myself, I should have left the `null` and `undefined` handling in AS3 `isXMLName` function. Can create in follow up PR.